lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -50,6 +51,7 @@ public AblyRealtime(ClientOptions options) throws AblyException {
 		super(options);
 		channels = new Channels();
 		connection = new Connection(this);
+		http.setHost(options.realtimeHost);
```

Comment:
You're missing the point. The host that you happen to be connected to will change from time to time, because the fallback hosts behaviour applies to realtime connections, not just rest requests. So it won't work to freeze the realtime host here for the lifetime of the library.